### PR TITLE
Use pipenv also for the default dependencies

### PR DESCRIPTION
**Pull request by nsoranzo**
Supersedes https://github.com/galaxyproject/galaxy/pull/4921.

### CODE REVIEW

File: lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+dictobj==0.4
+docopt==0.6.2
+docutils==0.14
+dogpile.cache==0.6.5
+enum34==1.1.6; python_version < '3'
+fabric3==1.14.post1
+funcsigs==1.0.2
+functools32==3.2.3.post2; python_version == '2.7'
+futures==3.2.0; python_version == '2.7' or python_version == '2.6'
+galaxy-sequence-utils==1.1.2
```
Comment (jmchilton) [inline lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt:37]: So pip just handles replacing - with _ magically it seems. Cool beans.

**Comment by jmchilton** [comment]
I don't know how you've caught h5py problem - wow - you clearly did a lot of manual testing for this and it is appreciated.

I'm +1 on this - it is really exciting. I'd have probably pinned Whoosh, python-genomespaceclient, and boto in the Pipfile to reflect that we don't have very great test coverage of any of features that make the most use of these I think. I also worry about indirect use of numpy but I don't have a sense of the scope of that. All that said - if something breaks down the road we can just re-pin these so I think this is absolutely fine and probably better than I would have managed to do for its boldness.

**Comment by nsoranzo** [comment]
Thanks @jmchilton!

Ping @nuwang about the python-genomespaceclient pinning.

**Comment by nuwang** [comment]
@nsoranzo genomespace lib reference seems ok to me.

**Comment by dannon** [comment]
@nsoranzo I just checked this branch out to test and cycled my Galaxy, but it looks like it's rebuilding local wheels for everything instead of using prebuilt.  Is that intentional?

**Comment by nsoranzo** [comment]
@dannon There are a lot of updated and new dependencies in this PR for which we would need to build wheels on Starforge (or convince upstream to do). My thought is we could update the dependencies at each start of a release cycle (as here) and build the needed wheels before release. 

Does this make sense? If agreed, we can modify the `RELEASE_ISSUE_TEMPLATE` in `scripts/bootstrap_history.py` .

**Comment by dannon** [comment]
@nsoranzo Ok, but the plan is definitely to prebuild and provide wheels, at least for releases -- cool.  I just wanted to make sure of that.  This all seems great to me, and the sooner we get it in during the dev cycle, the more likely we are to find any issues :)

**Comment by dannon** [comment]
@nsoranzo Looks like the pysam bump unveiled a problem w/ local installs -- some users won't have appropriate headers for at least lzma.  Maybe we should pin this back to 0.14.(1?), since this one is trickier to compile and won't 'just work' on a lot of OSX systems?  I'm creating a new 0.14.1 wheel right now, but I don't think we can expect to roll a new wheel every time there's a minor update.

(to document a quick fix for affected folks `brew install htslib` should fix a local pysam build, but we can't expect naive users to do that)

**Comment by nsoranzo** [comment]
@dannon Sorry for the trouble and thanks for the fast Starforge build!
We can either pin or annotate the problematic packages in the `Pipfile`, so we don't update them without a preliminary Starforge wheel.
And naive users shouldn't probably run the`dev` branch.

**Comment by dannon** [comment]
@nsoranzo Naive was probably a poor choice of a word (though galaxy at least *starting* for naive users on dev should also work, IMO) -- this was not me, but another core contributor having problems running Galaxy after an update.  I'll open a PR to pin pysam to the new wheel I made so that we're only 'floating' dependencies that reliably compile without extra packages.

**Comment by natefoo** [comment]
Thanks for all your work on dependencies lately @nsoranzo!

**Pull request merged**